lib/plugins/helpers/docker.rb

```diff
@@ -100,6 +99,25 @@ def hash_set(filename = nil, script = nil)
     end
   end
 
+  def expose
+    container_port = @container_hash[:ExposedPorts].keys[0].to_s
+    ip = @config['plugin']['public'] ? '0.0.0.0' : Socket.ip_address_list.detect(&:ipv4_private?).ip_address
+    @port = @config['plugin']['exposed_port'].blank? ? @dynamic_port : @config['plugin']['exposed_port']
```

Comment:
There are a lot of instance variables that likely should be attr_reader, attr_writer, or attr_accessor as appropriate.  Setting instance variables for the first time outside of the initialize is dangerous as complexity is added to code.

I believe this `port` is one of these cases.